stripped/colabs/Regresja_logistyczna_i_SVM_stripped.py

```diff
@@ -0,0 +1,1180 @@
+
+nasa_full_2019 = spark.sql('''SELECT * from nasa_full where Year=2019''')
+nasa_full_2019.show(5)
+nasa_full_2021 = spark.sql('''SELECT * from nasa_full where Year=2021''')
+nasa_full_2019.show(5)
+
+nasa_2019_an = nasa_full_2019.join(anotated_coordinates, on=['lon', 'lat'], how='inner').toPandas()
+nasa_2021_an = nasa_full_2021.join(anotated_coordinates, on=['lon', 'lat'], how='inner').toPandas()
+
+January_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==1])
+January_2019_DF.reset_index(inplace=True, drop=True)
+
+January_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==1])
+January_2021_DF.reset_index(inplace=True, drop=True)
+
+May_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==5])
+May_2019_DF.reset_index(inplace=True, drop=True)
+
+May_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==5])
+May_2021_DF.reset_index(inplace=True, drop=True)
+
+July_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==7])
+July_2019_DF.reset_index(inplace=True, drop=True)
+
+July_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==7])
+July_2021_DF.reset_index(inplace=True, drop=True)
+
+August_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==8])
+August_2019_DF.reset_index(inplace=True, drop=True)
+
+August_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==8])
+August_2021_DF.reset_index(inplace=True, drop=True)
+
+December_2019_DF = pd.DataFrame(data = nasa_2019_an[nasa_2019_an['Month']==12])
+December_2019_DF.reset_index(inplace=True, drop=True)
+
+December_2021_DF = pd.DataFrame(data = nasa_2021_an[nasa_2021_an['Month']==12])
+December_2021_DF.reset_index(inplace=True, drop=True)
+
+#Funkcja do generowania podziału na foldy do crosswalidacji blokowej (spatial block crossvalidation)
+#kod ze sprintu 1 musiałam zaadaptować pod Pandasa
+
+
+def get_grid(grid_cell_size: float, min_lat: float, max_lat: float,
+             min_lon: float, max_lon: float) -> Tuple[List[float], List[float]] :
+ ''' grid_cell_size - the grid cell size (the length of the side of
+                         a square) in degrees '''
+
+ area = (max_lat-min_lat)*(max_lon-min_lon)
+ cells_num = area//(grid_cell_size*grid_cell_size)
+ actual_grid_size = math.sqrt(area/cells_num)
+
+ xx=np.arange(min_lon, max_lon, step = actual_grid_size)
+ yy=np.arange(min_lat, max_lat, step = actual_grid_size)
+
+ return xx, yy
+
+def block_partition(df: pd.DataFrame, block_size: float, min_lat: float,
+                    max_lat: float, min_lon: float, max_lon: float) -> pd.DataFrame :
+  ''' block_size: approximate size of a block '''
+
+  blocked_df = copy.deepcopy(df)
+  block_number = np.zeros(len(df))
+  xx, yy = get_grid(block_size, min_lat, max_lat, min_lon, max_lon)
+
+  block_count = 0
+  for y in range(len(yy)-1):
+    for x in range(len(xx)-1):
+      block = df[(xx[x]<=df['lon']) & (df['lon'] <= xx[x+1]) & ( yy[y]<= df['lat']) & (df['lat'] <= yy[y+1])]
```

Comment:
To jest taki problem kiedy staramy się nadto uprościć strukturę w pamieci. Wymusza na tobie posługiwanie się indexem w liście `(range(len(yy))`. Ale python i na to ma sposoby
```suggestion
  for y_low, y_high in zip(yy[:-1], yy[1:]):
    for x_low, x_high in zip(xx[:-1], xx[1:]):
      block = df[(x_low<=df['lon']) & (df['lon'] <= x_high) & ( y_low<= df['lat']) & (df['lat'] <= y_high)]
```

Reply:
"Wymusza na tobie posługiwanie się indexem w liście (range(len(yy))" - czyli nie wolno się nim posługiwać? Jakiego typu tu jest problem, bo fajnie byłoby wiedzieć? ;)

Reply:
Operacje z indeksami są bardzo nieczytelne. Dlaczego pętla kończy się jeden wcześniej. Dlaczego akurat bierzemy x+1 element. etc. 
A tu masz inaczej od razu. Parujemy element z jego następnym i przeglądamy kolejne pary. Pierwszy nazywamy low, drugi high. Czyli pewnie wyznaczaja zakres. A okey w kodzie filtrujemy zmienne by w niego wpadały.

Reply:
Jeśli w kodzie napiszesz
```
x_low >= df['lon']
```
to czytając problem wypatrzysz niemal natychmiast.
Jeśli napiszesz
```
xx[x] >= df['lon']
```
wszystko wygląda poprawnie.

Reply:
OK.